src/app/components/server/StatusPill.tsx

```diff
@@ -24,71 +24,85 @@ export const StatusPillTypeMap: Record<string, StatusPillType> = {
   Fixed: "fixed",
 };
 
-type DirectTypeProps = { type: StatusPillType };
-type ExposureProps = { exposure: Exposure };
-export type Props = (DirectTypeProps | ExposureProps) & {
+export type Props = {
+  directType?: StatusPillType;
   enabledFeatureFlags?: FeatureFlagName[];
   note?: string;
+  exposure?: Exposure;
+  isRemovalUnderMaintenance?: boolean;
 };
 
 // This component just renders HTML without business logic:
 /* c8 ignore start */
 export const StatusPill = (props: Props) => {
   const l10n = useL10n();
-  const pillType = hasDirectType(props)
-    ? props.type
-    : getExposureStatus(
-        props.exposure,
-        props.enabledFeatureFlags?.includes("AdditionalRemovalStatuses") ??
-          false,
-      );
+
+  // Determine pill type and status label based on props
+  const pillType = props.directType
+    ? props.directType
+    : props.exposure
+      ? getExposureStatus(
+          props.exposure,
+          props.enabledFeatureFlags?.includes("AdditionalRemovalStatuses") ??
+            false,
+          props.isRemovalUnderMaintenance || false,
```

Comment:
Rather that using `||` we should explicitly check if the prop is `undefined`:

```suggestion
          typeof props.isRemovalUnderMaintenance !== "undefined",
```